### PR TITLE
Gpudevice: Return all displays as reference.

### PR DESCRIPTION
**Pull request by harishkrupo**
Jira: None
Test: No regressions seen on Android and Linux.

Signed-off-by: Harish Krupo <harish.krupo.kps@intel.com>

### CODE REVIEW

**Comment by strassek** [comment]
Why would we want to do this? Please write a commit message.

**Comment by kalyankondapally** [comment]
thx merged

**Comment by kalyankondapally** [comment]
breaks CI

**Comment by harishkrupo** [comment]
libtool: link: ar cru .libs/libjson-c.a  .libs/arraylist.o .libs/debug.o .libs/json_c_version.o .libs/json_object.o .libs/json_object_iterator.o .libs/json_pointer.o .libs/json_tokener.o .libs/json_util.o .libs/json_visit.o .libs/linkhash.o .libs/printbuf.o .libs/random_seed.o
libtool: link: ranlib .libs/libjson-c.a
libtool: link: ( cd ".libs" && rm -f "libjson-c.la" && ln -s "../libjson-c.la" "libjson-c.la" )
Making all in tests
Makefile:809: *** missing separator.  Stop.
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2

Checking...

**Comment by harishkrupo** [comment]
@kalyankondapally done.

File: common/core/gpudevice.cpp [hunk 1]
```diff
@@ -95,7 +95,7 @@ void GpuDevice::GetConnectedPhysicalDisplays(
   }
 }
 
-std::vector<NativeDisplay *> GpuDevice::GetAllDisplays() {
+std::vector<NativeDisplay *> &GpuDevice::GetAllDisplays() {
```
Comment (kalyankondapally) [inline common/core/gpudevice.cpp]: make this const std::vector<NativeDisplay *> &, so no one accidentally manipulates it.

**Comment by kalyankondapally** [comment]
ya, we should probably figure out a better way to expose this displays to platform layers. While doing that, this probably is better than what we have now.

**Comment by harishkrupo** [comment]
Hmm something wrong with the build :(
E: Unable to locate package g++-4.9
E: Couldn't find any package by glob 'g++-4.9'
E: Couldn't find any package by regex 'g++-4.9'
The command "sudo apt-get install automake autoconf libtool xutils-dev libpciaccess-dev python-mako bison flex libomxil-bellagio-dev libexpat1-dev llvm-dev gcc-4.9 g++-4.9 python3 python3-pip libudev-dev libmtdev-dev mtdev-tools libevdev-dev libx11-xcb-dev libxkbcommon-dev libxrandr-dev x11proto-*-dev libxcb* libxdamage* libx11-xcb-dev libxext-dev libxshmfence-dev libwacom-dev libgtk-3-dev check libpam0g-dev clang-format-3.9 cppcheck/trusty-backports" failed and exited with 100 during .

**Comment by harishkrupo** [comment]
@kalyankondapally

**Pull request closed without merging**